bin/utest/time.c

```diff
@@ -10,14 +11,74 @@ int test_gettimeofday(void) {
   for (int g = 0; g < 100; g++) {
     assert(gettimeofday(&time1, NULL) == 0);
     assert(gettimeofday(&time2, NULL) == 0);
+
     /* Time can't move backwards */
     assert(timercmp(&time2, &time1, <) == 0);
+
     /* Time belong to XXI century */
     assert(time1.tv_sec >= start_of_century);
     assert(time1.tv_sec <= end_of_century);
     assert(time2.tv_sec >= start_of_century);
     assert(time2.tv_sec <= end_of_century);
   }
 
+  return 0;
+}
+
+int test_nanosleep(void) {
+  /* Requested and remaining time */
+  timespec_t rqt, rmt;
+  timeval_t time1, time2, diff;
+  int ret;
+
+  /* Incorrect arguments */
+  rqt.tv_sec = -1;
+
+  rqt.tv_nsec = 1;
+  ret = nanosleep(&rqt, NULL);
+  assert(ret == -1 && errno == EINVAL);
+  ret = nanosleep(&rqt, &rmt);
+  assert(ret == -1 && errno == EINVAL);
+
+  rqt.tv_sec = 0;
+  rqt.tv_nsec = -1;
+  ret = nanosleep(&rqt, NULL);
+  assert(ret == -1 && errno == EINVAL);
```

Comment:
nit: imo it would be cleaner to use some macro like `assert_fail(expr, err)` (similar is used in other tests).

Reply:
Sure! Didn't noticed that solution before!